ansible/roles/shell/scripts/gm-mcp-ln.py

```
@@ -0,0 +1,87 @@
+
+def sync_mcp_servers(start_dir: Path) -> tuple[Dict[str, Any], Path]:
+    project_root = _find_project_root(start_dir)
+    mcp_path = project_root / ".mcp.json"
+    gemini_settings = start_dir / ".gemini" / "settings.json"
+
+    mcp_content = _read_json(mcp_path)
+    servers = mcp_content.get("mcpServers")
+    if servers is None:
+        raise ValueError(f"No mcpServers key found in {mcp_path}")
+    if not isinstance(servers, dict):
+        raise ValueError("mcpServers must be an object")
```

Ruff (0.13.2)

Prefer `TypeError` exception for invalid type

(TRY004)